### PR TITLE
test: Convert filter explain tests to new explain setup

### LINKED ISSUE #1379: Convert basic default explain tests including filter tests.
Convert basic default explain tests including filter tests.

More info in parent ticket: https://github.com/sourcenetwork/defradb/issues/953

### PR DESCRIPTION
**Pull request by shahzadlone**
## Relevant issue(s)
- Part of #953
- Resolves #1379

## Description
Continue converting explain tests to the new explain setup before we can integrate the entire setup to the new action based setup. #953 Has a lot more detail on the entire plan.

- Removes example tests that we don't need any more, if there is anything in them documentation wise that a reviewer wants found helpful I can introduce a README.md upon request.
- Removes some tests that provide no value wrt to non-execution explain tests.

### CODE REVIEW

**Comment by source-devs** [comment]
## Benchmark Results

### Summary
* 0 Benchmarks successfully compared.
* 0 Benchmarks were ✅ Better.
* 0 Benchmarks were ❌ Worse .
* 0 Benchmarks were ✨ Unchanged.


<details>
<summary> ✅ See Better Results...</summary>
<table class='benchstat better'>
<tbody>
<tr><th><th colspan='2' class='metric'>time/op<th>delta

<tr><td>&nbsp;
</tbody>
</table>
</details>

<details>
<summary> ❌ See Worse Results...</summary>
<table class='benchstat worse'>
<tbody>
<tr><th><th colspan='2' class='metric'>time/op<th>delta

<tr><td>&nbsp;
</tbody>
</table>
</details>

<details>
<summary> ✨ See Unchanged Results...</summary>
<table class='benchstat unchanged'>
<tbody>
<tr><th><th colspan='2' class='metric'>time/op<th>delta

<tr><td>&nbsp;
</tbody>
</table>
</details>

<details>
<summary> 🐋 See Full Results...</summary>

</details>

File: tests/integration/explain/utils.go [hunk 1]
```diff
@@ -123,14 +123,23 @@ func ExecuteExplainRequestTestCase(
...
+		explainTest.ExpectedFullGraph == nil {
+		require.Fail(t, "Atleast one expected explain parameter must be provided.", explainTest.Description)
+	}
+
+	// If we expect an error, then all other expected results should be empty (they shouldn't be provided).
+	if explainTest.ExpectedError != "" &&
+		(explainTest.ExpectedFullGraph != nil ||
+			explainTest.ExpectedPatterns != nil ||
+			explainTest.ExpectedTargets != nil) {
+		require.Fail(t, "Expected error should not have other expected results with it.", explainTest.Description)
```
Comment (AndrewSisley) [inline tests/integration/explain/utils.go:142]: :laughing: :+1:

**Review by AndrewSisley — APPROVED** [review]
Looks good to me :)

**Comment by shahzadlone** [comment]
For some odd reason I don't know why CodeCov report isn't showing. I really want to ensure code coverage doesn't change.

**Pull request merged**